src/Stack/Package.hs

```diff
@@ -1108,43 +1141,45 @@ findCandidate dirs exts name = do
 -- | Warn the user that multiple candidates are available for an
 -- entry, but that we picked one anyway and continued.
 warnMultiple
-    :: MonadLogger m
+    :: (MonadLogger m, HasRunner env, MonadReader env m)
     => DotCabalDescriptor -> Path b t -> [Path b t] -> m ()
 warnMultiple name candidate rest =
-    $logWarn
-        ("There were multiple candidates for the Cabal entry \"" <>
-         showName name <>
-         "\" (" <>
-         T.intercalate "," (map (T.pack . toFilePath) rest) <>
-         "), picking " <>
-         T.pack (toFilePath candidate))
-  where showName (DotCabalModule name') = T.pack (D.display name')
-        showName (DotCabalMain fp) = T.pack fp
-        showName (DotCabalFile fp) = T.pack fp
-        showName (DotCabalCFile fp) = T.pack fp
+    -- TODO: figure out how to style 'name' and the dispOne stuff
+    $prettyWarnL
+        [ flow "There were multiple candidates for the Cabal entry \""
+        , fromString . showName $ name
+        , line <> bulletedList (map dispOne rest)
+        , line <> flow "picking:"
+        , dispOne candidate
+        ]
+  where showName (DotCabalModule name') = D.display name'
+        showName (DotCabalMain fp) = fp
+        showName (DotCabalFile fp) = fp
+        showName (DotCabalCFile fp) = fp
+        dispOne = fromString . toFilePath
 
 -- | Log that we couldn't find a candidate, but there are
 -- possibilities for custom preprocessor extensions.
 --
 -- For example: .erb for a Ruby file might exist in one of the
 -- directories.
 logPossibilities
-    :: (MonadIO m, MonadThrow m, MonadLogger m)
+    :: (MonadIO m, MonadThrow m, MonadLogger m, HasRunner env,
+        MonadReader env m)
     => [Path Abs Dir] -> ModuleName -> m ()
 logPossibilities dirs mn = do
     possibilities <- liftM concat (makePossibilities mn)
-    case possibilities of
-        [] -> return ()
-        _ ->
-            $logWarn
-                ("Unable to find a known candidate for the Cabal entry \"" <>
-                 T.pack (D.display mn) <>
-                 "\", but did find: " <>
-                 T.intercalate ", " (map (T.pack . toFilePath) possibilities) <>
-                 ". If you are using a custom preprocessor for this module " <>
-                 "with its own file extension, consider adding the file(s) " <>
-                 "to your .cabal under extra-source-files.")
+    unless (null possibilities) $ $prettyWarnL
+        [ flow "Unable to find a known candidate for the Cabal entry"
+        , (styleModule . fromString $ D.display mn) <> ","
+        , flow "but did find:"
+        , line <> bulletedList (map dispPoss possibilities)
+        , flow "If you are using a custom preprocessor for this module"
+        , flow "with its own file extension, consider adding the file(s)"
+        , flow "to your .cabal under extra-source-files."
+        ]
   where
+    dispPoss = styleFile . fromString . toFilePath
```

Comment:
I think `display` can be used instead because there is an instance for `Path t File`.